onnxruntime/core/framework/utils.cc

```diff
@@ -450,7 +450,7 @@ static common::Status CopyInputsAcrossDevices(const SessionState& session_state,
                                               gsl::span<const OrtValue> orig_feeds,
                                               std::vector<OrtValue>& new_feeds,
                                               gsl::span<const MLValueCopyInfo> copy_info,
-                                              std::vector<Stream*>& feed_streams) {
+                                              gsl::span<Stream*> feed_streams) {
```

Comment:
>gsl::span<Stream*> feed_streams) [](http://example.com/codeflow?start=46&length=32)

If this is an input parameter, then it should be `gsl::span<Stream* const>`

Reply:
The element Stream* is then invoking its member function stream->Flush() which doesn't decorate as "const". So if we use gsl::span<Stream* const> here, there will be compile errors.

This is something I want to confirm: gsl::span can only be used when the parameter cannot be modified (if the parameter is a pointer, then neither the pointer itself nor the object the pointer is pointing to should be unchanged). For this case, if stream->Flush() is not changing the stream object, it should be decorated as "const"

Otherwise we should not use gsl::span. Is my understanding correct?